### PR TITLE
Feature/login

### LINKED ISSUE #19: [FEATURE] Login
## Feature Description

Ability to log in

## Definition of Done

- [x] Login page, responsive
- [x] Form validation
- [x] Error handling
- [x] Save tokens

### PR DESCRIPTION
**Pull request by TunixR**
# :warning: You need to have the auth microservice up to date to test this PR :warning: 

This pull request introduces several new components and features, updates dependencies, and makes various improvements to the codebase. The most important changes include adding a login page with form handling, creating reusable UI components, and updating dependencies in `package.json`.

### New Components and Features:
* Added a `Login` page with a `LoginForm` component that includes form validation and submission handling using `react-hook-form` and `zod`. [[1]](diffhunk://#diff-c0ba5a0f79600a67aa7a9bd6dd7e292a0d8cb1819c4ec9604c49d55975e6b803R1-R97) [[2]](diffhunk://#diff-0aade667e9a50cda16b0211fdb48cb4fe94ad6d9d6342c46e511f312bb738054R1-R67) [[3]](diffhunk://#diff-ecfc4ed960e754b1fb50fba2eb7fc2e45ac17127f8058a991f84b3098497cd4dL3)
* Created a `PasswordInput` component that allows users to toggle password visibility.
* Implemented a `PulsingIcons` component that displays animated icons.
* Added new UI components including `Card`, `Form`, and `Label` with various subcomponents for building forms and cards. [[1]](diffhunk://#diff-04ae3458baf6dac280dbee01019249280a78adac6439c6b7c48380c7dc19ef48R1-R61) [[2]](diffhunk://#diff-5f717a1fb79f908df342f224f2598b3f1f35215cffa8000b25f1cc85e21eebdeR1-R142) [[3]](diffhunk://#diff-a8120adafcde63b3464f359835b00ca096d2b501aae0d38c97fadee0f30bd0e4R1-R16)

### Dependency Updates:
* Updated `package.json` to include new dependencies: `@hookform/resolvers`, `react-hook-form`, and `zod`. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14-R19) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R30-R34)

### Code Improvements:
* Updated `ModeToggle` component to improve the transition between light and dark modes.
* Added animations for pulsing and error states in `animations.css`.

These changes enhance the application's functionality, improve user experience, and ensure better code organization and maintainability.

### CODE REVIEW

File: src/pages/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        else if (response.status === 200 && response.data.message === 'Credentials validated, please verify 2FA token')
+          navigate('/verify-2fa');
+      })
+      .catch((err) => {
+        if (err.response && err.response.status === 400) {
+          // Usually we would use form.setError, but in this case the 400 error only happens for "invalid credetials" which has no target field
+          setError('Invalid email or password');
+        } else {
+          setError('An error occurred. Please try again later.');
```
[the remark this answers is not in the document]
Comment (TunixR) [reply src/pages/Login.jsx:70]: NO

File: src/pages/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        <LoginForm form={form} onSubmit={onSubmit} isLoading={isLoading} error={error} />
+      </CardContent>
+    </Card>
+  );
+}
+
+function setIconsRed() {
+  document.querySelectorAll('.pulse').forEach((div) => {
+    div.classList.add('error-pulse');
+    div.classList.remove('pulse');
```
[the remark this answers is not in the document]
Comment (TunixR) [reply src/pages/Login.jsx:96]: This is not a form error but just a class change, there is no state implemented in that component.

**Comment by alvgonfri** [comment]
Shouldn't we have an issue for this task?

**Comment by alvgonfri** [comment]
I was getting a CORS error when trying to log in:
image

To make it work, I had to add the following CORS configuration to `api.js` file in the authorization microservice:
`app.use(cors({ origin: process.env.CORS_ORIGIN, credentials: true }));`

**Comment by TunixR** [comment]
> I was getting a CORS error when trying to log in: image
> 
> To make it work, I had to add the following CORS configuration to `api.js` file in the authorization microservice: `app.use(cors({ origin: process.env.CORS_ORIGIN, credentials: true }));`

This is not enough to make it work when the gateway is introduced and the host are differents. The fix for this is on the api-gateway.

Moreover, there is also the issue of cookies not being accepted on CORS, so I also had to add SameSite=Lax to the cookies sent by the auth service

**Review by alvgonfri — APPROVED** [review]
Works correctly!

**Pull request merged**